File: examples/python/best-practices/good-parallel-operations.py
@durable_execution
def lambda_handler(event: dict, context: DurableContext) -> dict:
    # Execute all three operations concurrently
    results = context.parallel(
        fetch_user_data(event["user_id"]),
        fetch_order_history(event["user_id"]),
        fetch_preferences(event["user_id"]),
    )
    
    return {
        "user": results[0],
        "orders": results[1],
        "preferences": results[2],
    }

File: examples/java/best-practices/good-parallel-operations.java
var userId = (String) input.get("user_id");
// Execute all three operations concurrently
DurableFuture<Map> userFuture = ctx.stepAsync("fetch_user_data", Map.class,
    stepCtx -> fetchUserData(userId));
DurableFuture<Map> ordersFuture = ctx.stepAsync("fetch_order_history", Map.class,
    stepCtx -> fetchOrderHistory(userId));
DurableFuture<Map> prefsFuture = ctx.stepAsync("fetch_preferences", Map.class,
    stepCtx -> fetchPreferences(userId));

return Map.of(
    "user", userFuture.get(),
    "orders", ordersFuture.get(),
    "preferences", prefsFuture.get()
);
